synapse/third_party_rules/access_rules.py

```diff
@@ -557,13 +558,24 @@ def _get_members_and_tokens_from_state(state_events):
         """
         existing_members = []
         threepid_invite_tokens = []
-        for key, event in state_events.items():
-            if key[0] == EventTypes.Member:
-                existing_members.append(event.state_key)
-            if key[0] == EventTypes.ThirdPartyInvite:
-                threepid_invite_tokens.append(event.state_key)
+        for key, state_event in state_events.items():
+            if key[0] == EventTypes.Member and state_event.content:
+                existing_members.append(state_event.state_key)
+            if key[0] == EventTypes.ThirdPartyInvite and state_event.content:
+                threepid_invite_tokens.append(state_event.state_key)
+
+        # If the event is a state event, there already is an event with the same state key
```

Comment:
very long sentence is exceedingly lengthy.

and tbh it doesn't really explain *why* this is the right thing to do